charactersheet/charactersheet/models/common/presence.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+/**
+An object representation of an XMPP presence message.
+
+This class provides a number of different convenience methods for routing, and
+reasoning about presence messages.
+*/
+function Presence() {
+    var self = this;
+    self.ps = PersistenceService.register(Presence, self);
+    self.mapping = {
+        include: ['hasParticipantRole', 'hasModeratorRole', 'hasNoneRole', 'from', 'hasError', 'dateReceived']
+    };
+
+    self.messageType = ko.observable(CHAT_MESSAGE_TYPES.SYSTEM);
+
+    // Model Methods
+
+    self.clear = function() {
+        var values = new Presence().exportValues();
+        var mapping = ko.mapping.autoignore(self, self.mapping);
+        ko.mapping.fromJS(values, mapping, self);
+    };
+
+    self.importValues = function(values) {
+        var mapping = ko.mapping.autoignore(self, self.mapping);
+        ko.mapping.fromJS(values, mapping, self);
+    };
+
+    self.exportValues = function() {
+        var mapping = ko.mapping.autoignore(self, self.mapping);
+        return ko.mapping.toJS(self, mapping);
+    };
+
+    self.save = function() {
+        self.ps.save();
+    };
+
+    self.delete = function() {
+        self.ps.delete();
+    };
+
+    // From Methods
+
+    self.fromNick = function() {
+        return Strophe.getResourceFromJid(self.from());
+    };
+
+    self.fromBare = function() {
+        return Strophe.getBareJidFromJid(self.from());
+    };
+
+    // Party/Room Methods
+
+    self.regardsCurrentParty = function() {
```

Comment:
I'm using it as in: "Does the information I provide regard the current party."

Is that clear or should we change it?